pennylane/pauli/pauli_arithmetic.py

```diff
@@ -377,7 +378,12 @@ def operation(self, wire_order=None):
         return summands[0] if len(summands) == 1 else qml.sum(*summands)
 
     def hamiltonian(self, wire_order=None):
-        """Returns a native PennyLane :class:`~pennylane.Hamiltonian` representing the PauliSentence."""
+        """Returns a native PennyLane :class:`~pennylane.Hamiltonian` representing the PauliSentence.
+
+        .. warning::
+            Any complex components of the coefficients will be set to real as a PennyLane hamiltonian cannot have
```

Comment:
well... yes they can

Reply:
I know, but physically this shouldn't be allowed. Also I was getting a weird error with gradients since the expectation value was complex in type, it was saying we need to use a holomorphic gradient.

Long story short: I removed this, I adjusted the failing test as it wasn't really trying to test complex coefficients in a hamiltonian anyways. Everything should work as expected. 🚀